kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/Utils.java

```diff
@@ -89,6 +90,62 @@ static void verifySuccess(TStatus status, boolean withInfo) throws SQLException
     throw new KyuubiSQLException(status);
   }
 
+  /**
+   * Splits the parametered sql statement at parameter boundaries.
+   *
+   * <p>taking into account ' and \ escaping.
+   *
+   * <p>output for: 'select 1 from ? where a = ?' ['select 1 from ',' where a = ','']
+   */
+  static List<String> splitSqlStatement(String sql) {
+    List<String> parts = new ArrayList<>();
+    int apCount = 0;
+    int off = 0;
+    boolean skip = false;
+
+    for (int i = 0; i < sql.length(); i++) {
+      char c = sql.charAt(i);
+      if (skip) {
+        skip = false;
+        continue;
+      }
+      switch (c) {
+        case '\'':
+          apCount++;
+          break;
+        case '\\':
+          skip = true;
+          break;
+        case '?':
+          if ((apCount & 1) == 0) {
+            parts.add(sql.substring(off, i));
+            off = i + 1;
+          }
+          break;
+        default:
+          break;
+      }
+    }
+    parts.add(sql.substring(off, sql.length()));
+    return parts;
+  }
+
+  /** update the SQL string with parameters set by setXXX methods of {@link PreparedStatement} */
+  public static String updateSql(final String sql, HashMap<Integer, String> parameters)
```

Comment:
it seems hack to me. How does Trino update parameters ?

Reply:
oh I see, shall we unifiy them ? can we make `org.apache.kyuubi.jdbc.hive.KyuubiPreparedStatement` use the updateSql from `Utils`

Reply:
sure，I think so too